File: service/src/main/java/hu/bsstudio/team/model/DetailedTeam.java
package hu.bsstudio.team.model;

import hu.bsstudio.entity.TeamType;
import java.util.List;
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

@Value
@Builder
@Jacksonized
public class DetailedTeam {
    long teamId;
    int year;
    String teamName;
    List<String> teamMembers;
    TeamType teamType;
    int technicalScore;
    int speedScore;
    int speedBonusScore;
    int numberOfOvertakes;
    boolean safetyCarWasFollowed;
    List<Integer> speedTimes;
    int votes;
    int audienceScore;
    int qualificationScore;
    int totalScore;
    int rank;
    int juniorRank;
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/team/model/DetailedTeam.kt
package hu.bsstudio.robonaut.team.model

import hu.bsstudio.robonaut.entity.TeamType

data class DetailedTeam(
  val teamId: Long = 0,
  val year: Int = 0,
  val teamName: String = "",
  val teamMembers: List<String> = listOf(),
  val teamType: TeamType = TeamType.JUNIOR,
  val skillScore: Int = 0,
  val numberOfOvertakes: Int = 0,
  val safetyCarWasFollowed: Boolean = false,
  val speedTimes: List<Int> = listOf(),
  val votes: Int = 0,
  val audienceScore: Int = 0,
  val qualificationScore: Int = 0,
  val combinedScore: Score = Score(),
  val juniorScore: Score = Score(),
)
